PrMerger.js

```diff
@@ -0,0 +1,92 @@
+const Config = require('./Config.js');
+const Log = require('./Logger.js');
+const Logger = Log.Logger;
+const GH = require('./GitHubUtil.js');
+const Util = require('./Util.js');
+const MergeContext = require('./MergeContext.js');
+
+// Gets PR list from GitHub and processes some/all PRs from this list.
+class PrMerger {
+
+    constructor() {
+        this.total = 0;
+        this.errors = 0;
+        // stores the the number of milliseconds to be re-run
+        // for the oldest 'slow burner'
+        this.rerunIn = null;
+    }
+
+    // Gets PR list from GitHub and processes them one by one.
+    // Returns if either all PRs have been processed(merged or skipped), or
+    // there is a PR still-in-merge.
+    async runStep() {
+        Logger.info("runStep running");
+        if (await this._resumeCurrent())
+            return true; // still in-process
+
+        const prList = await GH.getPRList();
+        prList.sort((pr1, pr2) => { return new Date(pr1.created_at) - new Date(pr2.created_at); });
```

Comment:
I doubt PR creation dates are guaranteed to be unique. They are also not as easily accessible as PR numbers. PR numbers are probably unique. Let's use PR numbers/IDs instead.

Reply:
Done.